docs/nodes/nodes-library/nodes/GoogleSheets/README.md

```diff
@@ -34,39 +34,137 @@ The final workflow should look like the following image.
+12. Click on the gears icon next to the ***Value*** field click on ***Add Expression***.
+13. Paste the following expression `{{Math.floor(Math.random()*1000)}}`. This expression will generate a three-digit random number.
+14. Toggle ***Keep Only Set*** to true.
+15. Click on ***Execute Node*** to run the node.
+:::
 
+![Using the Set node to set data to be inserted in the Google Sheets node](./Set_node.png)
 
+  
 ### 3. Google Sheets node (Append)
 
+This node will append the data in the Google Sheet.
+
 1. First of all, you'll have to enter credentials for the Google Sheets node. You can find out how to do that [here](../../../credentials/Google/README.md), in the section 'Google Drive / Sheets API'.
 2. Select 'OAuth2' in the ***Authentication*** field.
 3. Select 'Append' from the ***Operation*** dropdown list.
 4. Copy the string of characters located between `/d/` and `/edit` in your spreadsheet URL. Paste that string in the ***Sheet ID*** field.
-5. In the ***Range*** field, enter the range of columns to append the data to in your spreadsheet. Make sure that your range includes enough columns for all the data included in the Set node.
-6. Enter `id` in the A1 cell and `name` in the B1 cell  of the spreadsheet.
-7. Click on ***Execute Node*** to run the workflow.
+5. In the ***Range*** field, enter the range of columns to append the data to, in your spreadsheet. Make sure that your range includes enough columns for all the data in the Set node. For this workflow, enter `A:D` in the ***Range*** field.
+6. Click on the ***Add Option*** button and select 'Value Input Mode' from the dropdown list.
+7. Select 'User Entered' from the ***Value Input Mode*** dropdown list.
+8. In your spreadsheet, enter `ID`, `Name`, `Rent`, and `City` in the cells A1, B1, C1, and D1 respectively.
+9. Click on ***Execute Node*** to run the workflow.
 
 ![Using the Google Sheets node to insert data into a Google Sheets spreadsheet](./GoogleSheets_node.png)
 
+### 4. Google Sheets1 node (Lookup)
+
+This node will return the entries that have `Berlin` as the city.
+
+1. Select the credentials that you entered in the previous Google Sheets node.
+2. Select 'OAuth2' in the ***Authentication*** field.
+3. Select 'Lookup' from the ***Operation*** dropdown list.
+4. In the ***Sheet ID*** field, enter the same string used in the previous Google Sheets node.
+5. Enter `City` in the ***Lookup Column*** field.
+6. Enter `Berlin` in the ***Lookup Value*** field.
+7. Click on the ***Add Option*** button and select 'Return All Matches' from the dropdown list.
+8. Toggle ***Return All Matches*** to true.
+9. Click on the ***Add Option*** button and select 'Value Render Mode' from the dropdown list.
+10. Click on ***Execute Node*** to run the workflow.
+
+![Using the Google Sheets node to lookup for data in the Google Sheets spreadsheet](./GoogleSheets1_node.png)
+
+::: v-pre
+### 5. Set1 node
+
+This node will increase the rent by $100 for the houses in Berlin.
```

Comment:
Perhaps good to mention that this data is passed in the workflow by the previous node and that we'll use expression to do that.